internal/manifests/targetallocator/configmap.go

```diff
@@ -68,10 +68,20 @@ func ConfigMap(params manifests.Params) (*corev1.ConfigMap, error) {
 		prometheusCRConfig["scrape_interval"] = params.OtelCol.Spec.TargetAllocator.PrometheusCR.ScrapeInterval.Duration
 	}
 
+	prometheusCRConfig["service_monitor_selector"] = &metav1.LabelSelector{
```

Comment:
just to confirm my understanding here... by not having the nil check we follow what prometheus expects i.e. when service/pod monitor selector is nil nothing will be selected? whereas before we just didn't set it and assumed everything?

Reply:
> just to confirm my understanding here... by not having the nil check we follow what prometheus expects i.e. when service/pod monitor selector is nil nothing will be selected? whereas before we just didn't set it and assumed everything?

Prometheus' (and LabelSelector in general) behaviour is that `nil` == `nothing` and `empty` == `everything`. We on the other hand, took the map being empty or nil to mean `everything`, and there wasn't any way to set it to be `nothing`.

I've modified the configmap tests (and the builder and reconciliation tests) to show the empty label selectors. E2E tests check that this works as well - I know because I had a bug here earlier, and they caught it.

Reply:
In a sense, I'm moving the logic from target allocator here. If you look at the target allocator changes, it used to have this for the Prometheus Spec:

```go
ServiceMonitorSelector: &metav1.LabelSelector{
	MatchLabels: cfg.ServiceMonitorSelector,
},
```

Now we have this here, accomplishing the same thing.

Reply:
gotcha, thanks for explaining.